src/core/meshcop/panid_query_client.cpp

```diff
@@ -71,9 +71,8 @@ otError PanIdQueryClient::SendQuery(uint16_t                            aPanId,
     VerifyOrExit(Get<MeshCoP::Commissioner>().IsActive(), error = OT_ERROR_INVALID_STATE);
     VerifyOrExit((message = MeshCoP::NewMeshCoPMessage(Get<Tmf::TmfAgent>())) != nullptr, error = OT_ERROR_NO_BUFS);
 
-    SuccessOrExit(error =
-                      message->Init(aAddress.IsMulticast() ? OT_COAP_TYPE_NON_CONFIRMABLE : OT_COAP_TYPE_CONFIRMABLE,
-                                    OT_COAP_CODE_POST, OT_URI_PATH_PANID_QUERY));
+    SuccessOrExit(error = message->Init(aAddress.IsMulticast() ? Coap::kTypeNonConfirmable : Coap::kTypeConfirmable,
```

Comment:
There are 3 places that `Init` message according to if the address is multicast. Worth to define a method for that?

Reply:
Good idea. I noticed the same but was at first thinking that passing the `aAddress` to the `Message::Init()` may not be good/clear. Thinking more about it, I feel it can be fine (it's the destination based on which the CoAP type (confirmable vs non-confirmable is determined).